mobile/src/app/modules/profile/profile.page.ts

```diff
@@ -1,171 +1,104 @@
-  }
-
-  set address(address: string) {
-    this.profileForm.get('address').setValue(address);
-  }
-
-  get address(): string {
-    return this.profileForm.get('address').value;
-  }
-
-  set phone(phone: string) {
-    this.profileForm.get('phone').setValue(phone);
-  }
-
-  get phone(): string {
-    return this.profileForm.get('phone').value;
-  }
-
-  set isAvailable(isAvailable: boolean) {
-    this.profileForm.get('isAvailable').setValue(isAvailable);
-  }
-
-  get isAvailable(): boolean {
-    return this.profileForm.get('isAvailable').value;
-  }
-
-  ngOnInit() {
-  }
-
-  ionViewWillEnter() {
-    this.profileService.getProfile().subscribe(profile => {
-      if (profile) {
-        this.populateProfileForm(profile);
-        this.hasProfile = true;
-      } else {
-        this.hasProfile = false;
-      }
-    });
-  }
-
-  public async saveProfile() {
-    const profile = this.getProfileFromForm();
+  ) {
+    this.isDarkMode = window.matchMedia('(prefers-color-scheme: dark)').matches;
+    this.img = `assets/img/abbiamo-bisogno-del-tuo-aiuto${this.isDarkMode? '-dark' : ''}.png`;
+   }
+
+   ionViewWillEnter() {
+     this.profileService.getProfile().subscribe( p => {
+       if(!!p) {
+        this.user = p;
+       } else {
+         this.user = new Profile();
+         console.log(this.user.isHelper);
+       }
+     });
+   }
+
+   private async saveProfile(): Promise<void> {
     const loading = await this.loadingCtrl.create({
       message: this.translateConfigService.translateInstant('PROFILE_PAGE.LOADER_MESSAGE'),
       spinner: 'crescent',
     });
     loading.present();
-
-    this.profileService.addProfile(profile);
-    if (this.isAvailable) {
-      this.profileService.publishProfile(profile);
+    this.profileService.addProfile(this.user);
+    if (this.user.published) {
+      this.profileService.publishProfile(this.user);
     } else {
-      this.profileService.unpublishProfile(profile);
+      this.profileService.unpublishProfile(this.user);
     }
 
     loading.dismiss();
   }
-
-  public messageSubmit(): string {
-    return this.hasProfile ? 'SAVE' : 'SIGN_UP';
+  
+  public async registrati(): Promise<void> {
+    if(this.user.isProfileValid()) {
+      try{
+        await this.saveProfile();
+        this.router.navigate(['/']);
+      } catch(e) {
+        this.alertCtrl.create({
+          message: 'Siamo spiacenti, si è verificato un errore',
+          buttons: [
+            { text: 'Ho capito' }
+          ]
+        }).then( a => a.present());
+      }
+    } else {
+      this.showToast();
+    }
   }
 
-  public goToHome() {
-    this.navCtrl.navigateRoot('home/tabs/tab1');
+  public indietro(): void {
+    this.router.navigate(['/attivazione']);
   }
 
-  public setPosition() {
-    this.router.navigate(['position-piker']);
+  public openLocationSelector(): void {
+    this.modal.create({component: LocationSelectionPage}).then( m => m.present());
```

Comment:
riusciamo qui a passare il profilo alla pagina che popola la position e restituisce indietro l'oggetto?

Reply:
sì, è una modale standard in stile ionic 3, vediamo se far modificare la posizione per referenza o quando si torna indietro